### PR TITLE
chore: Daily Habits Tab Render Enhancements

### PR DESCRIPTION
**Pull request by calebpanza**
## ✏️ Description
This PR attempts to resolve 2 issues found while testing: lagging in the UI for completion states of each experience & overlapped renderings of the HeroCard as the user progresses from state-to-state

These issues have proven difficult to test because their replication steps are inconsistent and not easily reproduced. Using educated guesses, the following enhancements have been made:

1. Optimistic cache updates when the user submits a Gratitude
2. Optimistic cache updates when the user prays for a Prayer Request
3. Optimistic cache updates when the user navigates to the Scripture Experience for the given day
4. The HeroCard now uses a rendering _function_ wrapped in the `useCallback` hook in order to better tell the system that the whole of the section should be re-rendered on data changes
5. Each of the sections within the Daily Tab are given a unique Key

## 🔬 To Test
Like I said, the replication steps are inconsistent. Navigate through the Daily tab as though you were making an attempt at completing all 3 experiences to finish "today". Take note of re-render speeds/consistencies and also lack of visual artifacts

## 📸 Screenshots
n/a

## ⚠️ To-do Before Merge
n/a

## 📝 Checklist

- [x] I have performed a self-review of my code
- [x] I have tested my code works in the project (nothing breaks)
- [x] I have commented my code in hard-to-understand areas
- [x] New and existing tests and linting pass locally with my changes
- [x] I have made corresponding changes to the documentation (new envs?)
- [x] For PR's with design updates/changes, I have consulted with the product designer and made all necessary adjustments in Figma

### CODE REVIEW

File: templates/mobile/src/components/DailyHabitsTab/DailyHabitsTab.js [hunk 1]
```diff
@@ -31,6 +38,10 @@ export const useDailyHabitsTab = () => {
 const DailyHabitsTab = ({}) => {
```
Comment (dsilva609) [inline templates/mobile/src/components/DailyHabitsTab/DailyHabitsTab.js:38]: just noticed this, is the empty object actually nessessary here?

**Comment by jordanwade** [comment]
Is this PR fixing this issue? 
image

**Comment by jordanwade** [comment]
Just got this error. I wonder if it is related. 
image

File: templates/mobile/src/components/DailyHabitsTab/DailyHabitsTab.js (hunk 1 above)
Comment (calebpanza) [reply templates/mobile/src/components/DailyHabitsTab/DailyHabitsTab.js:38]: Nope! Lol

**Comment by calebpanza** [comment]
@jordanwade are you seeing that error on the DailyHabitsTab or on the Content Item?

**Comment by dsilva609** [comment]
> Just got this error. I wonder if it is related. image

I see this error too when loading up a scripture

**Comment by calebpanza** [comment]
@jordanwade @dsilva609 pushed up some new changes. Can you test it out for me?

**Comment by dsilva609** [comment]
> 

@calebpanza looks like it's behaving better now 💪

**Review by dsilva609 — APPROVED** [review]

**Comment by jordanwade** [comment]
Is there a way to reset my progress locally? Do I need to kill something in the DB? I think the transition from start to continue is where I was seeing the overlapping problem.

**Pull request merged**